static/app/views/explore/multiQueryMode/hooks/useMultiQueryTable.tsx

```diff
@@ -0,0 +1,122 @@
+import {useMemo} from 'react';
+
+import type {NewQuery} from 'sentry/types/organization';
+import EventView from 'sentry/utils/discover/eventView';
+import type {Sort} from 'sentry/utils/discover/fields';
+import {DiscoverDatasets} from 'sentry/utils/discover/types';
+import {MutableSearch} from 'sentry/utils/tokenizeSearch';
+import usePageFilters from 'sentry/utils/usePageFilters';
+import {formatSort} from 'sentry/views/explore/contexts/pageParamsContext/sortBys';
+import {getFieldsForConstructedQuery} from 'sentry/views/explore/multiQueryMode/locationUtils';
+import {useSpansQuery} from 'sentry/views/insights/common/queries/useSpansQuery';
+
+type Props = {
+  enabled: boolean;
+  groupBys: string[];
+  query: string;
+  sortBys: Sort[];
+  yAxes: string[];
+};
+
+export function useMultiQueryTableAggregateMode({
+  groupBys,
+  query,
+  yAxes,
+  sortBys,
+  enabled,
+}: Props) {
+  const {selection} = usePageFilters();
+
+  const fields = useMemo(() => {
+    const allFields: string[] = [];
+
+    for (const groupBy of groupBys) {
+      if (allFields.includes(groupBy)) {
+        continue;
+      }
+      allFields.push(groupBy);
+    }
+
+    for (const yAxis of yAxes) {
+      if (allFields.includes(yAxis)) {
+        continue;
+      }
+      allFields.push(yAxis);
+    }
```

Comment:
If im understanding this correctly, I think you could use a `Set` here? So you don't have to check for existence and just drop everything into an `allFields` `Set` blindly.